v-next/core/test/internal/tasks/builders.ts

```diff
@@ -0,0 +1,1157 @@
+import assert from "node:assert/strict";
+import { describe, it } from "node:test";
+
+import { expectTypeOf } from "expect-type";
+
+import {
+  NewTaskDefinitionBuilderImplementation,
+  TaskOverrideDefinitionBuilderImplementation,
+} from "../../../src/internal/tasks/builders.js";
+import { ParameterType } from "../../../src/types/common.js";
+import {
+  NewTaskDefinition,
+  TaskDefinitionType,
+  TaskOverrideDefinition,
+} from "../../../src/types/tasks.js";
+
+describe("Task builders", () => {
+  describe("NewTaskDefinitionBuilderImplementation", () => {
+    it("should create a new task definition builder", () => {
+      const builder = new NewTaskDefinitionBuilderImplementation("task-id");
+      const taskAction = () => {};
+      const taskDefinition = builder.setAction(taskAction).build();
+
+      expectTypeOf(
+        builder,
+      ).toEqualTypeOf<NewTaskDefinitionBuilderImplementation>();
```

Comment:
I don't think I'd do `expectTypeOf` test unless there's some advanced typing like the one you did with the default values. The reason is that the type system is implicitly doing the same test. e.g. the deepEqual below woudln't type.